docs/writing-rules/experiments/pattern-syntax.md

```diff
@@ -0,0 +1,253 @@
+
+There is often a one-to-one translation from the existing syntax to the experimental syntax. These changes are marked with <i class= "fa-solid fa-diamond"></i>. However, some changes are quite different. These changes are marked with <i class="fa-solid fa-exclamation"></i>
+
+:::warning
+These patterns are **experimental** and subject to change.
+
+You can't mix and match existing pattern syntax with the experimental syntax.
+:::
+
+## <i class="fa-solid fa-exclamation"></i> `pattern`
+
+The `pattern` operator looks for code matching its expression in the existing syntax. However, `pattern` is no longer required when using the experimental syntax. For example, you can use `...` wherever `pattern: "...``` appears. For example, you can omit `pattern` and write the following:
+
+```yaml
+any:
+  - "badthing1"
+  - "badthing2"
+  - "badthing3"
+```
+
+or, for multi-line patterns
+
+```yaml
+any:
+  - |
+      manylines(
+        badthinghere($A)
+      )
+  - |
+      orshort()
+```
+
+You don't need double quotes for a single-line pattern when omitting the `pattern` key, but note that this can cause YAML parsing issues.
+
+As an example, the following YAML parses:
+
+```yaml
+any:
+  - "def foo(): ..."
+```
+
+This, however, causes problems since `:` is also used to denote a YAML dictionary:
+
+```yaml
+any:
+  - def foo(): ...
+```
+
+### <i class="fa-solid fa-diamond"></i> `any`
+
+Replaces [pattern-either](/writing-rules/rule-syntax/#pattern-either). Matches any of the patterns specified.
+
+```yaml
+any:
+  - <pat1>
+  - <pat2> 
+    ...
+  - <patn>
+```
+
+### <i class="fa-solid fa-diamond"></i> `all`
+
+Replaces [patterns](/writing-rules/rule-syntax/#patterns). Matches all of the patterns specified.
+
+```yaml
+all:
+  - <pat1>
+  - <pat2> 
+    ...
+  - <patn>
+```
+
+### <i class="fa-solid fa-diamond"></i> `inside`
+
+Replaces [pattern-inside](/writing-rules/rule-syntax/#pattern-inside). Match any of the sub-patterns inside of the primary pattern.
+
+```yaml
+inside: 
+  any: 
+    - <pat1> 
+    - <pat2>
+```
+
+Alternatively:
+
+```yaml
+any:
+  - inside: <pat1>
+  - inside: <pat2>
+```
+
+### <i class="fa-solid fa-diamond"></i> `not`
+
+Replaces [pattern-not](/writing-rules/rule-syntax/#pattern-not). Accepts any pattern and does **not** match on those patterns.
+
+```yaml
+not:
+  any:
+    - <pat1>
+    - <pat2>
+```
+
+Alternatively:
+
+```yaml
+all:
+  - not: <pat1>
+  - not: <pat2>
+```
+
+### <i class="fa-solid fa-diamond"></i> `regex`
+
+Replaces [pattern-regex](/writing-rules/rule-syntax/#pattern-regex) Matches based on the regex provided.
+
+```yaml
+regex: "(.*)"
+```
+
+## Metavariables
+
+Metavariables are an abstraction to match code when you don't know the value or contents beforehand. They're similar to [capture groups](https://regexone.com/lesson/capturing_groups) in regular expressions and can track values across a specific code scope. This
```

Comment:
(This nit is completely optional and can be ignored.) Do we want to standardize the definition of a metavariable?